src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -65,23 +81,30 @@ def get_data_batch(self, train, batch_size, seed=0):
         stop_i = start_i + batch_size
         shuffle_seed = (batch_size * seed) // len(train)
         np.random.seed(shuffle_seed)
-        train_ix = np.random.choice(list(train.index), replace=False, size=len(train))  # wasteful to shuffle every time
+        train_ix = np.random.choice(train.shape[0], replace=False, size=len(train))  # wasteful to shuffle every time
```

Comment:
it shouldn't be from shape, this will create problems if the index is not ordered. 

I would recommend to revert to index as this was an issue spotted in the past.

Reply:
Now I remember the reason for this change, after the preprocessing the object train is no longer a pd.DataFrame but a np.array.
I believe this change has no negative impact in get_data_batch functionality since we are using numpy's positional indexing instead of panda's loc index. For the purpose of shuffling it should not be affected by the index order. Let me know if I am missing something